packages/client/src/pages/Game/projectiles/Projectile.tsx

```diff
@@ -17,54 +16,129 @@ class Projectile {
+          this.tower.upgradeLevel
+        ] as HTMLCanvasElement[]
+      )[this.getNextFrameIndex()]!;
+      // get current frame to rotate projectile image and draw it in main projectile context
+      const context = (
+        this.tower.engine.towerSprites[this.tower.type]!.canvasContextArr
+          ?.projectile[this.tower.upgradeLevel]! as CanvasRenderingContext2D[]
+      )[this.tower.projectileParams.projectileFrameLimit]!;
+      // find rectangle diagonal
+      const canvasHypot = Math.ceil(
+        Math.hypot(
+          this.tower.engine.predefinedTowerParams[this.tower.type]
+            ?.projectileParams?.dimensions[this.tower.upgradeLevel]
+            .projectileWidth!,
+          this.tower.engine.predefinedTowerParams[this.tower.type]
+            ?.projectileParams?.dimensions[this.tower.upgradeLevel]
+            .projectileHeight!,
+        ),
       );
-    } else {
-      this.tower.engine.context!.strokeStyle = "black";
-      this.tower.engine.context?.strokeRect(
-        this.currentPosition.x,
-        this.currentPosition.y,
-        this.tower.projectileParams.width,
-        this.tower.projectileParams.height,
+      // rotate frame
+      context.save();
+      context.clearRect(0, 0, canvasHypot, canvasHypot);
+      context.beginPath();
+      context.translate(canvasHypot / 2, canvasHypot / 2);
+      context.rotate(this.tower.towerParams.firingAngle - 1);
+      context.drawImage(canvas, -canvasHypot / 2, -canvasHypot / 2);
+      context.closePath();
+      context.restore();
+      // and draw it
+      this.tower.engine.context!.projectile!.drawImage(
+        (
+          this.tower.engine.towerSprites[this.tower.type]!.canvasArr
+            ?.projectile![this.tower.upgradeLevel] as HTMLCanvasElement[]
+        )[3]!,
+        // this.tower.towerParams.fireFromCoords.x,
+        // this.tower.towerParams.fireFromCoords.y,
+        Math.ceil(this.currentPosition.x),
+        Math.ceil(this.currentPosition.y),
+      );
+    } else if (this.renderParams.isAnimateImpact) {
+      // impact
+      this.tower.engine.context!.projectile!.drawImage(
+        (
+          this.tower.engine.towerSprites[this.tower.type]!.canvasArr?.impact![
+            this.tower.upgradeLevel
+          ] as HTMLCanvasElement[]
+        )[this.getNextImpactFrameIndex()]! as HTMLCanvasElement,
+        Math.ceil(this.currentPosition.x),
+        Math.ceil(this.currentPosition.y),
+        /*
+        Math.ceil(
+          this.currentPosition.x -
+            this.tower.engine.predefinedTowerParams[this.tower.type]
+              ?.projectileParams?.dimensions[this.tower.upgradeLevel]
+              .impactWidth! /
+              2,
+        ),
+        Math.ceil(
+          this.currentPosition.y -
+            this.tower.engine.predefinedTowerParams[this.tower.type]
+              ?.projectileParams?.dimensions[this.tower.upgradeLevel]
+              .impactHeight! /
+              2,
+        ),
+         */
```

Comment:
Unexpected commented code without any annotation is a bad idea =-(

Reply:
This is actually a proper version of animation impact frame draw, but it's untested, so I ll just keep it here.

Anyway, before final release all dev or debug comments should be removed